internal/providers/pluginfw/tfschema/customizable_schema.go

```diff
@@ -199,8 +200,63 @@ func (s *CustomizableSchema) SetReadOnly(path ...string) *CustomizableSchema {
 	return s
 }
 
+// ConvertToAttribute converts the last element of the path from a block to an attribute.
+// It panics if the path is empty, if the path does not exist in the schema, or if the path
+// points to an attribute, not a block.
+func (s *CustomizableSchema) ConvertToAttribute(path ...string) *CustomizableSchema {
```

Comment:
This method is a bit complex. Can we simplify it somehow?

Just an idea: it looks like the parts related to `CustomizableSchema` are mostly outside the callback definition, in lines 207-210 and 247-255. What if we move the callback outside of this method? Would that make things easier to follow? Maybe it would be better to implement the callback within the builders and split it between `ListNestedBlockBuilder` and `SingleNestedBlockBuilder`?